samples/sample_camera/src/camera_sysmem_allocator.cpp

```diff
@@ -0,0 +1,380 @@
+        ptr->A = (mfxU8*)(ptr->V16 + 3);
+        ptr->PitchHigh = (mfxU16)((8 * (mfxU32)Width2) / (1 << 16));
+        ptr->PitchLow  = (mfxU16)((8 * (mfxU32)Width2) % (1 << 16));
+        break;
+    case MFX_FOURCC_ABGR16:
+        ptr->Y16 = (mfxU16*)ptr->B;
+        ptr->U16 = ptr->Y16 + 1;
+        ptr->V16 = ptr->Y16 + 2;
+        ptr->A = (mfxU8*)(ptr->Y16 + 3);
+        ptr->PitchHigh = (mfxU16)((8 * (mfxU32)Width2) / (1 << 16));
+        ptr->PitchLow  = (mfxU16)((8 * (mfxU32)Width2) % (1 << 16));
+        break;
+   case MFX_FOURCC_R16:
+        ptr->Y16 = (mfxU16 *)ptr->B;
+        ptr->Pitch = 2 * Width2;
+        break;
+    default:
+        return MFX_ERR_UNSUPPORTED;
+    }
+
+    return MFX_ERR_NONE;
+}
+
+mfxStatus CamSysMemFrameAllocator::UnlockFrame(mfxMemId mid, mfxFrameData *ptr)
+{
+    if (!m_pBufferAllocator)
+        return MFX_ERR_NOT_INITIALIZED;
+
+    mfxStatus sts = m_pBufferAllocator->Unlock(m_pBufferAllocator->pthis, mid);
+
+    if (MFX_ERR_NONE != sts)
+        return sts;
+
+    if (NULL != ptr)
+    {
+        ptr->Pitch = 0;
+        ptr->Y     = 0;
+        ptr->U     = 0;
+        ptr->V     = 0;
+    }
+
+    return MFX_ERR_NONE;
+}
+
+mfxStatus CamSysMemFrameAllocator::GetFrameHDL(mfxMemId mid, mfxHDL *handle)
+{
+    return MFX_ERR_UNSUPPORTED;
+}
+
+mfxStatus CamSysMemFrameAllocator::CheckRequestType(mfxFrameAllocRequest *request)
+{
+    mfxStatus sts = BaseFrameAllocator::CheckRequestType(request);
+    if (MFX_ERR_NONE != sts)
+        return sts;
+
+    if ((request->Type & MFX_MEMTYPE_SYSTEM_MEMORY) != 0)
+        return MFX_ERR_NONE;
+    else
+        return MFX_ERR_UNSUPPORTED;
+}
+
+mfxStatus CamSysMemFrameAllocator::AllocImpl(mfxFrameAllocRequest *request, mfxFrameAllocResponse *response)
+{
+    if (!m_pBufferAllocator)
+        return MFX_ERR_NOT_INITIALIZED;
+
+    mfxU32 numAllocated = 0;
+
+    mfxU32 Width2 = MSDK_ALIGN32(request->Info.Width);
+    mfxU32 Height2 = MSDK_ALIGN32(request->Info.Height);
+    mfxU32 nbytes;
+
+    switch (request->Info.FourCC)
+    {
+    case MFX_FOURCC_YV12:
+    case MFX_FOURCC_NV12:
+        nbytes = Width2*Height2 + (Width2>>1)*(Height2>>1) + (Width2>>1)*(Height2>>1);
+        break;
+    case MFX_FOURCC_RGB3:
+        nbytes = Width2*Height2 + Width2*Height2 + Width2*Height2;
+        break;
+    case MFX_FOURCC_RGB4:
+        nbytes = Width2*Height2 + Width2*Height2 + Width2*Height2 + Width2*Height2;
+        break;
+    case MFX_FOURCC_ARGB16:
+    case MFX_FOURCC_ABGR16:
+        nbytes = (Width2*Height2 + Width2*Height2 + Width2*Height2 + Width2*Height2) << 1;
+        break;
+    case MFX_FOURCC_YUY2:
+        nbytes = Width2*Height2 + (Width2>>1)*(Height2) + (Width2>>1)*(Height2);
+        break;
+    case MFX_FOURCC_R16:
+        nbytes = 2*Width2*Height2;
+        break;
+
+    default:
+        return MFX_ERR_UNSUPPORTED;
+    }
+
+    nbytes += m_alignment;
+
+    safe_array<mfxMemId> mids(new mfxMemId[request->NumFrameSuggested]);
```

Comment:
Windows build fails for me because safe_array is no longer defined (was removed by commit 4449756).

Reply:
Fixed